### PR TITLE
feat: Set EtcdDiskSizeGB max default size to 1023 if target platform is Azure Stack

### LINKED ISSUE #1557: feat: Set EtcdDiskSizeGB max default size to 1023 if target platform is Azure Stack
Set EtcdDiskSizeGB max default size to 1023 if target platform is Azure Stack as the max managed disk size supported on Azs is 1023GB.

### PR DESCRIPTION
**Pull request by rjaini**
**Reason for Change**:
Azure Stack only supports max disk size to be 1023GB. So in case when the number of nodes increases
we cant increase the size to 1024 or 2048GB.

**Issue Fixed**:
Fixes #1557 

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)  
- [ ] includes documentation
- [X] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/defaults.go [hunk 1]
```diff
@@ -233,9 +233,19 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 		if "" == a.OrchestratorProfile.KubernetesConfig.EtcdDiskSizeGB {
 			switch {
 			case a.TotalNodes() > 20:
-				a.OrchestratorProfile.KubernetesConfig.EtcdDiskSizeGB = DefaultEtcdDiskSizeGT20Nodes
+				if a.IsAzureStackCloud() {
+					// Currently on Azure Stack max size of managed disk size is 1023GB.
```
Comment (honcao) [inline pkg/api/defaults.go]: @rjaini do we know how much disk is required for 20 + nodes?  The cluster will be not usable if the disc is out of space.
Another alternative is to limit the total count of agent nodes.
And we should raise the request to Disk RP on Azure Stack to support larger size of disk.

Comment (rjaini) [reply pkg/api/defaults.go]: I am not sure how much disk space requirements. But from https://github.com/Azure/aks-engine/issues/923 thread it looks like the 1024 size itself is very high.
I will check with Disk RP team to understand the reason of 1023 limit.

@mboersma : Do you know why we have such large disk size for higher number of nodes in AKS Engine ?.

File: pkg/api/const.go [hunk 2]
```diff
@@ -252,6 +252,9 @@ const (
 
 	// DefaultAzureStackFaultDomainCount set to 3 as Azure Stack today has minimum 4 node deployment.
 	DefaultAzureStackFaultDomainCount = 3
+
+	// DefaultAzureStackEtcdDiskSizeGT10Nodes = size for Kubernetes master etcd disk volumes in GB if > 10 nodes
+	DefaultAzureStackEtcdDiskSizeGT10Nodes = "1023"
```
Comment (CecileRobertMichon) [inline pkg/api/const.go]: the naming "GT10Nodes" doesn't really apply here since there's only one default size

File: pkg/api/defaults.go (hunk 1 above)
Comment (CecileRobertMichon) [reply pkg/api/defaults.go]: Matt is out on vacation but see https://github.com/Azure/acs-engine/pull/2435 for context.

File: pkg/api/const.go (hunk 2 above)
Comment (rjaini) [reply pkg/api/const.go]: My Bad. Updating name to MaxAzureStackManagedDiskSize to be consistent to another PR.

**Pull request merged**